npm/private/lifecycle/lifecycle-hooks.js

```diff
@@ -68,6 +69,46 @@ async function makeBins(nodeModulesPath, scope, segmentsUp) {
     }
 }
 
+// Helper from not exported by @pnpm/lifecycle
+async function checkBindingGyp(root, scripts) {
+    if (await exists(path.join(root, 'binding.gyp'))) {
+        scripts['install'] = 'node-gyp rebuild'
+    }
+}
+
+// Like runPostinstallHooks from @pnpm/lifecycle but also runs prepare hook which is notably missing
+// from that function and does not match what pnpm does on its install:
```

Comment:
It looks like the `runPostinstallHooks` method is only used by the `@pnpm/plugin-commands-rebuild` npm package in `pnpm` which is described as "Commands for rebuilding dependencies". Maybe it is intentionally ommitted since the function is named "PostinstallHooks" so it only runs "preinstall", "install" and "postinstall"